addons/account/models/account_move.py

```diff
@@ -3874,11 +3874,11 @@ def _get_move_display_name(self, show_ref=False):
         :return:            A string representing the invoice.
         '''
         self.ensure_one()
-        if self.env.context.get('name_as_amount_total'):
+        if self.env.context.get('name_as_amount_total') or self.env.context.get('name_with_move_type'):
```

Comment:
if we go though the condition with the name_with_move_type and that the state is not posted we just continue ? I mean the condition is just used here and not after, weird ? Maybe a simple comment to explain ?